cms/core/models/base.py

```diff
@@ -66,9 +66,13 @@ def has_equations(self) -> bool:
         """Checks if there are any equation blocks.
         Override in your specific Page class if the StreamField structure is different.
         """
-        if streamfield := getattr(self, self.content_field_name):
+        if streamvalue := getattr(self, self.content_field_name):
+            # run the check on the StreamBlock itself, if it supports it
+            if hasattr(streamvalue.stream_block, "has_equations"):
+                return bool(streamvalue.stream_block.has_equations(streamvalue))
+
             try:
-                return streamfield.first_block_by_name(block_name="equation") is not None
+                return streamvalue.first_block_by_name(block_name="equation") is not None
```

Comment:
@zerolab I feel like we might be better to remove this alternative option and just let the StreamBlock implementation be the canonical API. What do you think?

Reply:
Just a note - if we decided to do that and remove those methods from `BasePage`, then the `content_scripts.html` template would have to be modified too.
We wouldn't be able to simply do  `{% if page.has_equations %}` anymore and would have to filter out the relevant `StreamBlock` and call its own `has_....` function.

Reply:
Good shout. 68232d5

Reply:
@kacperpONS we're not removing the methods, just the fallback check. See 68232d5